### PR TITLE
drop support for MacOS 10.14 (Mojave)

### PR DESCRIPTION
**Pull request by jjergus**
Needed for https://github.com/facebook/hhvm/pull/8813, because the latest Folly doesn't compile on MacOS 10.14.

### CODE REVIEW

File: aws/hhvm1/lambdas/test.py [hunk 1]
```diff
@@ -143,12 +143,12 @@ def test_get_platforms_for_version(self):
       ['debian-10-buster', 'ubuntu-19.04-disco'],
     )
     # incompatible platforms are excluded
-    input = {'platforms': ['ubuntu-19.10-eoan', 'ubuntu-14.04-trusty']}
+    input = {'platforms': ['ubuntu-20.04-focal', 'ubuntu-14.04-trusty']}
```
Comment (jjergus) [inline aws/hhvm1/lambdas/test.py:146]: Some of these tests need manual updates from time to time, which is not great, but better than not having tests or having tests where too much stuff is mocked out, IMHO.

File: aws/hhvm1/lambdas/test.py [hunk 2]
```diff
@@ -826,7 +826,12 @@ def test_build_and_publish_macos(self):
       Config.macos_versions.keys()
     )
     self.assertEqual(activity.should_run(), True)
-    self.assertEqual(activity.task_env(), {})
+    self.assertEqual(
+      activity.task_env(),
+      {'PLATFORM': Config.macos_versions[macos1]}
+        if len(Config.macos_versions) == 1
+        else {}
```
Comment (jjergus) [inline aws/hhvm1/lambdas/test.py:833]: https://github.com/hhvm/packaging/blob/master/aws/hhvm1/lambdas/activities.py#L278

**Review by fredemmott — APPROVED** [review]
The ubuntu changes here also seem fine if they're intentional

**Comment by fredemmott** [comment]
How urgent is this? If we're doing a second post-HAM LTS, it'd be good to hold off a little longer

**Comment by jjergus** [comment]
Required for https://github.com/facebook/hhvm/pull/8813 which is not super urgent but would be nice to get in soon.

**Comment by fredemmott** [comment]
Sounds like that LTS is a bit too far out, we probably shouldn't wait :)

**Pull request merged**